gnuradio-runtime/include/gnuradio/high_res_timer.h

```diff
@@ -123,34 +124,30 @@ inline gr::high_res_timer_type gr::high_res_timer_tps(void)
 #endif
 
 ////////////////////////////////////////////////////////////////////////
-#ifdef GNURADIO_HRT_USE_MICROSEC_CLOCK
+#ifdef GNURADIO_HRT_USE_GENERIC_CLOCK
 inline gr::high_res_timer_type gr::high_res_timer_now(void)
 {
-    static const boost::posix_time::ptime epoch(boost::posix_time::from_time_t(0));
-    return (boost::posix_time::microsec_clock::universal_time() - epoch).ticks();
+    return std::chrono::duration<gr::high_res_timer_type, std::nano>(
+               std::chrono::steady_clock::now().time_since_epoch())
+        .count();
 }
 
 inline gr::high_res_timer_type gr::high_res_timer_now_perfmon(void)
 {
     return gr::high_res_timer_now();
 }
 
-inline gr::high_res_timer_type gr::high_res_timer_tps(void)
-{
-    return boost::posix_time::time_duration::ticks_per_second();
-}
+inline gr::high_res_timer_type gr::high_res_timer_tps(void) { return 1000000000UL; }
 #endif
 
 ////////////////////////////////////////////////////////////////////////
 inline gr::high_res_timer_type gr::high_res_timer_epoch(void)
 {
-    static const double hrt_ticks_per_utc_ticks =
-        gr::high_res_timer_tps() /
-        double(boost::posix_time::time_duration::ticks_per_second());
-    boost::posix_time::time_duration utc =
-        boost::posix_time::microsec_clock::universal_time() -
-        boost::posix_time::from_time_t(0);
-    return gr::high_res_timer_now() - utc.ticks() * hrt_ticks_per_utc_ticks;
+    static const double ticks_per_second = gr::high_res_timer_tps();
+    const double seconds_since_epoch =
+        std::chrono::duration<double>(std::chrono::system_clock::now().time_since_epoch())
+            .count();
```

Comment:
`system_clock` is used here because we need to measure time since the Unix epoch. The documentation states that prior to C++20 most implementations of `system_clock` used the Unix epoch, and as of C++20 implementations are required to do so. GNU Radio already depends on this behaviour in other places.